src/test/java/com/gildedrose/GildedRoseTest.java

```diff
@@ -189,5 +189,34 @@ void conjured_update_qualityDropsTwiceAsFastAsNormalItemsWhenSellInIsPassed0() {
         assertEquals(28, app.items[0].quality);
     }
 
+    @Test
+    void redWine_update_qualityDropsThreeTimesSlowerThanNormalItems() {
+        String name = "Red Wine";
+        int sellInDays = 7;
+        int quality = 32;
+        Item[] items = new Item[] { new Item(name, sellInDays, quality) };
+        GildedRose app = new GildedRose(items);
+
+        app.updateQuality();
+        app.updateQuality();
+        app.updateQuality();
+
+        assertEquals(31, app.items[0].quality);
+    }
+
+    @Test
+    void redWine_update_qualityDropsThreeTimesSlowerThanNormalItemsWhenSellInIsPassed0() {
+        String name = "Red Wine";
+        int sellInDays = 0;
+        int quality = 32;
+        Item[] items = new Item[] { new Item(name, sellInDays, quality) };
+        GildedRose app = new GildedRose(items);
+
+        app.updateQuality();
+        app.updateQuality();
+        app.updateQuality();
+
+        assertEquals(30, app.items[0].quality);
```

Comment:
```
int updatedQuality = 30
assertEquals(updatedQuality, app.items[0].quality)
```
Im on board with the similar suggestions mentioned above and below as well.

Reply:
hmm I actually find having the value shown directly instead of using a variable makes the test more readable and easier to reason about. Ideally something like:
```java
redWine_update_qualityDropsThreeTimesSlowerThanNormalItemsWhenSellInIsPassed0() {
  ...
  Item[] items = new Item[] { new Item(name: 'foo', sellInDays: 0, quality: '32') };
  ...
  assertEquals(30, app.item[0].quality)
}
```

Since the test is around the quality of an item, making that value more explicitly helps communicate the behavior we're looking for. When you add a variable, you're also adding a bit of indirection, especially in the case where that variable doesn't suggest duplication.

In general, I prefer duplication over abstraction when it comes to tests. I would only abstract details that aren't concerned with the test.

Reply:
Addressed in [this commit](https://github.com/jramnani/GildedRoseJavaLLC/pull/19/commits/ef0ba6e35361e945fb211ea3e0487af09982c377) (reverted the test variable assignments)